xcode/Subconscious/Shared/Components/Common/Byline/PetnameBylineView.swift

```diff
@@ -9,15 +9,32 @@ import SwiftUI
 
 /// Byline style for displaying a petname
 struct PetnameBylineView: View {
-    var petname: String
-    private var petnameColor = Color.accentColor
+    var petname: Petname
+    var petnameColor = Color.accentColor
     
     var body: some View {
-        Text(verbatim: "@\(petname)")
-            .font(.body)
-            .fontWeight(.medium)
-            .foregroundColor(petnameColor)
-            .lineLimit(1)
+        let parts = petname.markup.split(separator: ".")
+
+        HStack(alignment: .lastTextBaseline, spacing: 0) {
+            let first = parts[0]
+            let rest = parts[1...].joined(separator: ".")
```

Comment:
Not in love with having this logic inline here but it doesn't appear anywhere else (yet)